### PR TITLE
feat(ajv): add returnsCoercedValues option to keep the coerced value by ajv

### LINKED ISSUE #2497: [BUG] Body Validation not working as expected
### Describe the bug

I'm currently migrating a express application to Ts.ED. I used joi for the validation before and currently try to make AJV work, as i like the direct support & swagger integration.

I have a problem making the following work:

```ts
import { Integer, Required } from '@tsed/schema';

export class Test {
  @Integer()
  @Required()
  value: number;
}
```

```ts
@Post('/some-url')
async someFunction(@BodyParams() body: Test) {
  ...
}
```

Also writing `@BodyParams(Test) ...` does not work.

I don't find an exact documentation for this, except from `Any model used on parameter and annotated with one of JsonSchema decorator will be validated with AJV.` [written here](https://tsed.io/docs/validation.html). So i guess my usage is correct and should work?

Also in my `server.ts` i have:

```ts
import '@tsed/ajv';
```

What does work for me is something like:
```ts
@Post('/some-url')
async someFunction(@BodyParams('value') @Integer() @Required value: number) {
  ...
}
```
but i would like to validate the whole body. Is this possible?

### To Reproduce

Should be quite is to set up, see above.

Because this is also a kind of "should this be even possible" - issue, i dind't create an extra reproducable repo. If you need one, let me know.

### Expected behavior

I would expect the full body to be validated correctly when used as described above. The request should ail with the ajv error.

### Code snippets

_No response_

### Repository URL example

_No response_

### OS

macOS

### Node version

18.17.1

### Library version

7.41.2

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by EinfachHans**
## Information

| Type                  | Breaking change |
| --------------------- | --------------- |
| Fix | No ?          |

---

closes #2497

With this PR the result of a Validation is correctly returned. The resulted data can be different from the input data, when the validation modifies it, which is a common usage.

This is my first PR to this awesome project. Please let me know if i'm missing something 🤔

## Todos

- [x] Tests
- [x] Coverage
- [x] Example
- [x] Documentation

### CODE REVIEW

File: packages/specs/ajv/src/services/AjvService.ts [hunk 1]
```diff
@@ -44,6 +44,8 @@ export class AjvService {
           value: localValue
         });
       }
+
```
Comment (Romakita) [inline packages/specs/ajv/src/services/AjvService.ts:50]: We can remove the deepClone / localValue , give the value directly to ajv ;)

it’ll improve the performance!

**Review by Romakita — CHANGES_REQUESTED** [review]

**Comment by Romakita** [comment]
@EinfachHans thanks for the PR. Just a small change to optimize the code ;)

see you

**Comment by EinfachHans** [comment]
@Romakita okay somehow an integration test is now failing. Didn't know why it worked locally first...

Anyways, the test that fails is `Scenario8` in **testBodyParams.ts**, which tests a null model input should return the null model back. I'm unsure about this test, can you have a look?

**Comment by Romakita** [comment]
Yes I take a look. This test is here to check if the null value is keep through after validation and deserialization depending on the endpoint configuration.
This one of the side effect of ajv.
I’ll check if it’s pertinent or not.

**Comment by Romakita** [comment]
Ok I already given information on this issue and why isn't fixable easily:
https://github.com/tsedio/tsed/issues/2355

**Comment by EinfachHans** [comment]
Mhh, so should i close this PR?

**Review by Romakita — APPROVED** [review]

**Comment by Romakita** [comment]
@EinfachHans I fixed the problem by introducing a new ajv options `returnsCoercedValues`.

Can you review the latest commits please ;)

**Comment by Romakita** [comment]
This PR will solve your issue and another issue related to the coerced value by ajv! Good news :D

**Comment by EinfachHans** [comment]
@Romakita awesome, Thanks! 🚀
Code makes totally sense and documentation is quite clear or me 👍🏼

**Comment by Romakita** [comment]
Ok fixed issue on my local. It should be ok to merge and release after that ;)

**Pull request merged**